bin/dev

```diff
@@ -1,60 +1,71 @@
 #!/bin/bash
 
-# Find container id of syncm8_api in order to exec things against it
+DEV_CONTAINER_NAME="syncm8_dev"
+
+# Find container id of syncm8_dev in order to exec things against it
+# args: name of variable to set container id
 findBackendContainerId(){
     local  _resultvar=$1
-    local container=$(docker ps|grep syncm8_api|cut -d' ' -f1)
+    local container=$(docker ps|grep $DEV_CONTAINER_NAME|cut -d ' ' -f1)
     if [ ! $container ]; then
-        echo "No container for api found. Check that it is running with bin/dev ps, or start it with bin/dev up"
+        bin/util error "No container for dev found. Check that it is running with bin/dev ps, or start it with bin/dev up"
         exit 1
     fi
     eval $_resultvar="'$container'"
 }
 
-# Run docker compose commands as correct user
-runDockerCompose(){
-    user_id=$(id -u) group_id=$(id -g) docker-compose $*
+# Run commands in exiting dev container as correct user
+# args: commands to run in container
+runDockerExec(){
+    findBackendContainerId container_id
+    cmd="docker exec -it -u $(id -u):$(id -g) $container_id "$@
+    bin/util logAndRun $cmd
+}
+
+# Run commands with interactive turned off in exiting dev container
+# args: commands to run in container
+runDockerExecNonInteractive(){
```

Comment:
just curious, is there a clean way to refactor `runDockerExec` and `runDockerExecNonInteractive` into one?

Reply:
There is, but the shell syntax makes it really verbose, and then calling it also becomes a pain.